CollegiateSubredditsTable Component

isaacchin opened the pull request:
# Overview
Created CollegiateSubredditsTable component with columns for database fields 
Added fixtures for CollegiateSubredditsTable 
Added storybook entry for CollegiateSubredditsTable 
Added tests for CollegiateSubredditsTable with 100% scores with npm run coverage and npx stryker run

# Issues Addressed
#7 

# Details
Test coverage: ![image](https://user-images.githubusercontent.com/66229119/154876794-85fe99f0-65f7-4f2b-a400-d5275923d8b1.png)
Mutation testing: ![image](https://user-images.githubusercontent.com/66229119/154876811-8611ca36-fb64-4223-bc2f-ddeaefee9a23.png)


## Reminders (delete this section before merging or as code is merged)
- [x] Explicitly link to this pull request each issue that it closes
- [x] Assign one and ideally two reviewers to review this pull request. 
- [x] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

isaacchin committed:
Added CollegiateSubredditsTable component with related fixtures, tests, and storybook entry

mglush committed:
Merge branch 'main' into ic-subreddits-table

mglush commented:
resolved conflicts in the collegiate subreddit fixtures file.

mglush commented on frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js:
```diff
@@ -0,0 +1,157 @@
+  
+      expect(getByTestId(`${testId}-cell-row-0-col-id`)).toHaveTextContent("1");
+      expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("2");
+  
```
Why do you have 3 collegiate subreddits in your fixtures, load up 3 collegiate subreddits into the table, and then only check 1 and 2???

Add this on line 93:
expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("3");

isaacchin replied:
Good catch, I pushed the correction to the test now

mglush commented on frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js:
```diff
@@ -0,0 +1,59 @@
+        columns={memoizedColumns}
+        testid={"CollegiateSubredditsTable"}
+    />;
+};
```
all good here.

mglush commented on frontend/src/main/utils/CollegiateSubredditsUtils.js:
```diff
@@ -0,0 +1,18 @@
+        }
+    }
+}
+
```
all good here.

mglush commented on frontend/src/stories/components/CollegiateSubreddits/CollegiateSubredditsTable.stories.js:
```diff
@@ -0,0 +1,29 @@
+    subjects: collegiateSubredditsFixtures.threeSubreddits
+};
+
+
```
this file looks all good.

mglush commented on frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js:
```diff
@@ -0,0 +1,157 @@
+  
+  
+  });
+  
```
file requires some changes.

mglush reviewed (changes requested):
Need to change the test file to check the id of all 3 entries you loaded up, not just 2.

stychoi reviewed (commented):
Other than the thing Glush pointed out, LGTM

Commit:
mg - create page for UCSBSubjects now works, still have to write test coverage for the newly-written methods

Commit:
mg - wrote tests for npm coverage to be 100 and npx stryker to be 100. branch should now be ready for a PR

Commit:
added a dryRunTimeoutMin variable to stryker configuration so that the overall test timeout time is 45 minutes.

isaacchin committed:
Fixed CollegiateSubredditsTable test to test all three fixtures

Commit:
increased timeout

Commit:
catching branch up to merge
Merge branch 'ic-subreddits-table' of github.com:ucsb-cs156-w22/team03-w22-7pm-3 into ic-subreddits-table
